sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageInputStream.java

```diff
@@ -111,6 +111,33 @@ protected StorageInputStream(long rangeOffset, final Long rangeLength,
         this.reposition(rangeOffset);
     }
 
+    /**
+     * Initializes a new instance of the StorageInputStream class.
+     *
+     * @param rangeOffset The offset of the data to begin stream.
+     * @param rangeLength How much data the stream should return after blobRangeOffset.
+     * @param chunkSize Holds the stream read size.
+     * @param contentLength The length of the stream to be transferred.
+     * @throws IndexOutOfBoundsException when range offset is less than 0 or rangeLength exists but les than or
```

Comment:
Should this also throw if `rangeOffset + rangeLength` > `contentLength`?

Reply:
I don't think it needs to because of this line:

`        this.streamLength = rangeLength == null ? contentLength - this.rangeOffset
            : Math.min(contentLength - this.rangeOffset, rangeLength);`

So if the inequality you mentioned is true, we just cut it down to be the right size